spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -56,15 +68,61 @@ public DefaultSubscriberFactory(GcpProjectIdProvider projectIdProvider,
 		this.executorProvider = executorProvider;
 		this.channelProvider = channelProvider;
 		this.credentialsProvider = credentialsProvider;
+		this.pullTimeoutMillis = pullTimeoutMillis;
+	}
+
+	/**
+	 * Enables simple a timeout for subscription pull requests.
+	 *
+	 * @param pullTimeoutMillis Timeout in milliseconds for subscription pull requests. If
+	 * set, disables pull retry logic and uses the simple timeout. If set to null, default
+	 * pull behavior with retries will be used.
+	 */
+	public void setPullTimeoutMillis(Long pullTimeoutMillis) {
+		this.pullTimeoutMillis = pullTimeoutMillis;
 	}
 
 	@Override
-	public Subscriber getSubscriber(String subscriptionName, MessageReceiver receiver) {
+	public Subscriber createSubscriber(String subscriptionName, MessageReceiver receiver) {
 		return Subscriber.newBuilder(
 				SubscriptionName.of(this.projectId, subscriptionName), receiver)
 				.setChannelProvider(this.channelProvider)
 				.setExecutorProvider(this.executorProvider)
 				.setCredentialsProvider(this.credentialsProvider)
 				.build();
 	}
+
+	@Override
+	public PullRequest createPullRequest(String subscriptionName, Integer maxMessages,
```

Comment:
What happened to the idea of having a `createPullRequest()` method that takes in a timeout? :)

Reply:
I still think it makes sense to do that, but the Pub/Sub client library makes it awkward to implement because the `PullRequest` object does not have a timeout property. Instead timeout and retry configuration has to be configured on the `SubscriberStub`. Let me experiment with this more though.